### PR TITLE
Gh 2179 transformer pooling

### PR DESCRIPTION
**Pull request by kishaloyhalder**
Added 'mean', 'max' pooling strategy for `TransformerDocumentEmbeddings` class.

### CODE REVIEW

**Comment by kishaloyhalder** [comment]
Verified the correctness with sentence-transformers' implementation of the same.

```
from flair.embeddings.document import TransformerDocumentEmbeddings
from flair.data import Sentence
from sentence_transformers import SentenceTransformer, models
import torch

text = "Good Morning"
document_embeddings = TransformerDocumentEmbeddings(model="bert-base-uncased", pooling="mean")

# option 1
sentence = Sentence(text, use_tokenizer=False)
document_embeddings.embed(sentence)
embedding_1 = sentence.get_embedding()
print(embedding_1.shape)
print(embedding_1[:20])

# option 2
document_embeddings.pooling = 'max'
sentence = Sentence(text, use_tokenizer=False)
document_embeddings.embed(sentence)
embedding_2 = sentence.get_embedding()
print(embedding_2.shape)
print(embedding_2[:20])

# option 3
document_embeddings.pooling = 'cls'
sentence = Sentence(text, use_tokenizer=False)
document_embeddings.embed(sentence)
embedding_3 = sentence.get_embedding()
print(embedding_3.shape)
print(embedding_3[:20])


word_embedding_model = models.Transformer('bert-base-uncased', max_seq_length=512)
pooling_model = models.Pooling(word_embedding_model.get_word_embedding_dimension())

model = SentenceTransformer(modules=[word_embedding_model, pooling_model])

# option 1'
embedding_11 = model.encode(text, convert_to_tensor=True)
print(embedding_11.shape)
print(embedding_11[:20])

# option 2'
pooling_model.pooling_mode_cls_token = False
pooling_model.pooling_mode_mean_tokens = False
pooling_model.pooling_mode_max_tokens = True
embedding_21 = model.encode(text, convert_to_tensor=True)
print(embedding_21.shape)
print(embedding_21[:20])

# option 3'
pooling_model.pooling_mode_cls_token = True
pooling_model.pooling_mode_mean_tokens = False
pooling_model.pooling_mode_max_tokens = False
embedding_31 = model.encode(text, convert_to_tensor=True)
print(embedding_31.shape)
print(embedding_31[:20])

assert torch.all(torch.eq(embedding_1, embedding_11))
assert torch.all(torch.eq(embedding_2, embedding_21))
assert torch.all(torch.eq(embedding_3, embedding_31))
```

File: flair/embeddings/document.py [hunk 1]
```diff
@@ -159,20 +165,37 @@ def _add_embeddings_to_sentences(self, sentences: List[Sentence]):
...
 
                 # use scalar mix of embeddings if so selected
                 if self.layer_mean:
-                    sm = ScalarMix(mixture_size=len(cls_embeddings_all_layers))
-                    sm_embeddings = sm(cls_embeddings_all_layers)
+                    sm = ScalarMix(mixture_size=len(embeddings_all_layers))
+                    sm_embeddings = sm(embeddings_all_layers)
 
-                    cls_embeddings_all_layers = [sm_embeddings]
+                    embeddings_all_layers = [embeddings_all_layers]
```
Comment (alanakbik) [inline flair/embeddings/document.py:195]: This should be                     
```python
embeddings_all_layers = [sm_embeddings]
```

File: flair/embeddings/document.py [hunk 2]
```diff
@@ -50,9 +51,13 @@ def __init__(
         models tend to be huge.
         :param layers: string indicating which layers to take for embedding (-1 is topmost layer)
         :param layer_mean: If True, uses a scalar mix of layers as embedding
+        :param pooling: Pooling strategy for combining token level embeddings. options are 'cls', 'max', 'mean'.
         """
         super().__init__()
 
+        if pooling not in ['cls', 'max', 'mean']:
+            raise ValueError(f"Pooling operation `{poling}` is not defined for TransformerDocumentEmbeddings")
```
Comment (alanakbik) [inline flair/embeddings/document.py:59]: The variable `poling` does not exist ;) Should be `pooling`

**Review by alanakbik — CHANGES_REQUESTED** [review]

File: flair/embeddings/document.py (hunk 1 above)
Comment (kishaloyhalder) [reply flair/embeddings/document.py]: Done.

File: flair/embeddings/document.py (hunk 2 above)
Comment (kishaloyhalder) [reply flair/embeddings/document.py]: Done.

**Review by alanakbik — APPROVED** [review]

**Comment by alanakbik** [comment]
@kishaloyhalder thanks a lot for adding this!

**Pull request merged**